browser/android/youtube_script_injector/youtube_script_injector_tab_helper.cc

```diff
@@ -126,4 +121,46 @@ void YouTubeScriptInjectorTabHelper::PrimaryMainDocumentElementAvailable() {
   }
 }
 
+// static
+bool YouTubeScriptInjectorTabHelper::IsYouTubeVideo(const GURL& url) {
+  if (!url.is_valid() || url.is_empty()) {
+    return false;
+  }
+
+  // Check if domain is youtube.com (including subdomains).
+  if (!net::registry_controlled_domains::SameDomainOrHost(
+          url, GURL("https://www.youtube.com"),
```

Comment:
do we need `www` here?

Reply:
This part checks whether the given URL belongs to the YouTube domain or any of its subdomains (like `www.youtube.com`, `m.youtube.com`, `music.youtube.com`, etc.). 
The `www` should be irrelevant. I'm not sure if there are other reasons, as it's part of the original implementation that we have used for ages and it predates my time at Brave: https://github.com/brave/brave-core/commit/83e6513cfc1be0676c62cdc3b8042b06a903e99a
Happy to improve it if you suggestions 😀